addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animation_sampler_keyframes.py

```diff
@@ -62,12 +62,18 @@ def get_target_len(self):
         return length
 
     def __set_indexed(self, value):
+        if value is None:
+            return value
+        if "rotation" in self.target and type(value) is mathutils.Quaternion:
+            return value
         # 'value' targets don't use keyframe.array_index
         if self.target == "value":
             return value
         # Sometimes blender animations only reference a subset of components of a data target. Keyframe should always
         # contain a complete Vector/ Quaternion --> use the array_index value of the keyframe to set components in such
         # structures
+        if type(value) is mathutils.Quaternion:
```

Comment:
Not sure to understand why there is this code: the previous check added returns out the function for rotation/Quaternion. WHat is the case where value is a Quaternion and value is not "rotation" ?

Reply:
"rotation" targets are implemented in the plugin as Quaternions; thus, it makes sense in the first test to simply return the Quaternion. But if the target is not a "rotation",  code execution continues and resolves for a non-Quaternion target; thus, the second Quaternion test which returns: `tuple(value.to_euler("XYZ"))`